src/handlers/event-message-handler.ts

```diff
@@ -249,6 +261,10 @@ export class EventMessageHandler implements IMessageHandler {
       return
     }
 
+    if (this.getRelayPublicKey() === event.pubkey) {
+      return undefined
```

Comment:
Inconsistent returns, is this intentional?

Reply:
this is fine, we return `undefined` if user is admitted/whitelisted or a string with the reason if they are not admitted

this new check fixes the issue of the relay's public key not being allowed on the relay when payment is enabled which is a big oversight

Reply:
in the future we could change this pattern to using constants or symbols

Reply:
Ah, I'll remove the extra undefined